src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateVersions = [];
+  const stateVersion = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'removeProperties':
+        for (const keyToRemove of action.keysToRemove) {
+          if (Object.keys(stateVersion).includes(keyToRemove)) {
+            delete stateVersion[keyToRemove];
+          }
+        }
+        stateVersions.push({ ...stateVersion });
+        break;
+      case 'addProperties':
+        Object.assign(stateVersion, action.extraData);
+        stateVersions.push({ ...stateVersion });
+        break;
+      case 'clear':
+        for (const stateVersionKey in stateVersion) {
+          delete stateVersion[stateVersionKey];
+        }
+        stateVersions.push({ ...stateVersion });
```

Comment:
You can move the 'stateVersions.push' statement outside the switch block to reduce code duplication